src/common/itemSaver_background.js

```diff
@@ -230,15 +230,26 @@ Zotero.ItemSaver._fetchAttachment = async function(attachment, tab, attemptBotPr
 
 	// Bot bypass is not supported in Safari (cannot intercept file download popup)
 	attemptBotProtectionBypass = attemptBotProtectionBypass && !Zotero.isSafari;
-	var contentType;
+	var receivedMimeType;
 	try {
 		let xhr = await Zotero.HTTP.request("GET", attachment.url, options);
 		let result = Zotero.Utilities.Connector.getContentTypeFromXHR(xhr);
-		contentType = result.contentType;
+		// extract mimeType from contentType header by stripping the charset (if any)
+		receivedMimeType = result.contentType.split(';')[0]?.trim();
```

Comment:
Sorry, the diff before didn't include the lines above and I didn't notice we were already getting the content type from the response headers in the `getContentTypeFromXHR()`, which also extracts the charset separately and returns it. As such, this additional logic splitting the `;` isn't necessary.

I think we just need to check if attachment has a mime type, and if it doesn't, set it to `result.contentType`